scripts/test-check-ci-policy.py

```
@@ -0,0 +1,1970 @@
+                "main ordinary job must invoke the complete ordinary script",
+            ),
+        ):
+            with self.subTest(old=old):
+                self.write(
+                    ".github/workflows/ci-main.yml",
+                    VALID_MAIN_WORKFLOW.replace(old, "removed", 1),
+                )
+                self.assert_violation(expected)
+
+    def test_main_workflow_collects_only_logs_without_filename_collisions(self) -> None:
+        for old, expected in (
```

Ruff (0.16.1)

[warning] Loop control variable `expected` not used within loop body

Rename unused `expected` to `_expected`

(B007)